### ISSUE TITLE
purpose of try catch block in addImageToTemplate?

### ISSUE THREAD

**Issue by fritzmg** [contao/core-bundle#1479]
What is the purpose of
```php
try
{
    $objFile = new \File($arrItem['singleSRC']);
}
catch (\Exception $e)
{
    $objFile = new \stdClass();
    $objFile->imageSize = false;
}
```
at the beginning of [`Controller::addImageToTemplate`](https://github.com/contao/core-bundle/blob/4.5.7/src/Resources/contao/library/Contao/Controller.php#L1451-L1459)?

If the file does not exist for example, the code will fail [at the following point](https://github.com/contao/core-bundle/blob/4.5.7/src/Resources/contao/library/Contao/Controller.php#L1548-L1553) anyway:
```php
// Image dimensions
if ($objFile->exists() && ($imgSize = $objFile->imageSize) !== false)
{
    $objTemplate->arrSize = $imgSize;
    $objTemplate->imgSize = ' width="' . $imgSize[0] . '" height="' . $imgSize[1] . '"';
}
```
```
Attempted to call an undefined method named "exists" of class "stdClass".
```

**Comment by leofeyer** [comment]
@ausi /cc

**Comment by ausi** [comment]
I can’t remember. The related ticket is contao/core#7383 I think, but I don’t know why the call to `new \File()` should be in a `try` block.

**Comment by fritzmg** [comment]
In that PR there is no `$objFile->exists()` yet, so it would not fail at least.

**Comment by leofeyer** [comment]
As discussed in Mumble on May 17th, we have to check what the method does if the image does not exist. Then we have to make sure it does the same if the exception is thrown.

**Comment by leofeyer** [comment]
Ok, so here is what happens if the image does not exist:

```
Warning: filemtime(): stat failed for files/contaodemo/media/content-images/xDSC_5276.jpg
  at vendor/contao/core-bundle/src/Resources/contao/library/Contao/File.php:234
  at Contao\File->__get('mtime')
     (vendor/contao/core-bundle/src/Resources/contao/library/Contao/File.php:259)
  at Contao\File->__get('imageSize')
     (vendor/contao/core-bundle/src/Resources/contao/library/Contao/Controller.php:1435)
```

However, if `singleSRC` points to a directory:

```
Attempted to call an undefined method named "exists" of class "stdClass".
  at vendor/contao/core-bundle/src/Resources/contao/library/Contao/Controller.php:1521
```

Both cases are not handled appropriately if you ask me. 😄

**Comment by leofeyer** [comment]
As discussed in Mumble on August 30th, this is how we want to fix it:

```diff
diff --git a/core-bundle/src/Resources/contao/library/Contao/Controller.php b/core-bundle/src/Resources/contao/library/Contao/Controller.php
index 21ee45281..785b6e93e 100644
--- a/core-bundle/src/Resources/contao/library/Contao/Controller.php
+++ b/core-bundle/src/Resources/contao/library/Contao/Controller.php
@@ -1432,11 +1432,10 @@ abstract class Controller extends System
 		}
 		catch (\Exception $e)
 		{
-			$objFile = new \stdClass();
-			$objFile->imageSize = false;
+			$objFile = null;
 		}
 
-		$imgSize = $objFile->imageSize;
+		$imgSize = $objFile ? $objFile->imageSize : false;
 		$size = \StringUtil::deserialize($arrItem['size']);
 
 		if (is_numeric($size))
@@ -1524,7 +1523,7 @@ abstract class Controller extends System
 		}
 
 		// Image dimensions
-		if ($objFile->exists() && ($imgSize = $objFile->imageSize) !== false)
+		if ($objFile && $objFile->exists() && ($imgSize = $objFile->imageSize) !== false)
 		{
 			$objTemplate->arrSize = $imgSize;
 			$objTemplate->imgSize = ' width="' . $imgSize[0] . '" height="' . $imgSize[1] . '"';
```

**Comment by leofeyer** [comment]
Fixed in 25a7e87856e18a32ad8f4576929bed42d978b4b7.
